### ISSUE TITLE
Production phylesystem lagging (oti not being updated)

### ISSUE THREAD

**Issue by jimallman** [OpenTreeOfLife/phylesystem-1#4]
We pushed new code to production today. Apart from a missing neo4j plugin (fixed, but not truly explained), the site is looking good, with one exception — the [webhooks that trigger oti updates](https://github.com/OpenTreeOfLife/phylesystem-1/settings/hooks/2674614) haven’t been called since Jan 10 (about 2 weeks). 

I'm guessing the api is trying to push to github, but it's being blocked by unresolved conflicts or somesuch, reminiscent of https://github.com/OpenTreeOfLife/phylesystem-api/issues/106. How can I confirm this? (I don't see anything obvious in `api.log` or `peyotl.log`.)

Here's the state of the local `phylesystem-1` repo on **api**, with four commits not yet pushed to github:

``` bash
(venv)opentree@ip-10-21-133-161:~/repo/phylesystem-1_par/phylesystem-1$ git status
# On branch master
# Your branch is ahead of 'origin/master' by 4 commits.
#
nothing to commit (working directory clean)
(venv)opentree@ip-10-21-133-161:~/repo/phylesystem-1_par/phylesystem-1$ git log
commit 8e54ea98ddd511380c8f6819fc3813d8ab9e9edb
Author: Jim Allman <jim@ibang.com>
Date:   Fri Jan 23 21:40:40 2015 +0000

    tag test (oti is lagging)

    (Update Study #ot_54 via OpenTree API)

commit 3f4966ceebac7c8e6fd59db595bf4b4a9f2e0e01
Author: Jim Allman <jim@ibang.com>
Date:   Fri Jan 23 20:40:39 2015 +0000

    test of modified tag (oti refresh)

    (Update Study #ot_54 via OpenTree API)

commit e6095ad16bf31cedab3160dcf04b41c1d5b06ee0
Author: Jim Allman <jim@ibang.com>
Date:   Fri Jan 23 20:38:19 2015 +0000

    Update Study #ot_238 via OpenTree API

commit 7df13ef7824f31f4a9010f04a50fa4778d986294
Author: OpenTree API <api@opentreeoflife.org>
Date:   Fri Jan 23 20:38:18 2015 +0000

    {"next_study_id": 239}

commit 823d2aff31d3095dce04696ccc3a8e0d4d58c0b8
Author: Mark T. Holder <mtholder@gmail.com>
Date:   Sat Jan 10 20:47:13 2015 +0000

    rooting and branch length info

    this is tree1 BEAST exponential node dating with calibration densities.

    (Update Study #ot_237 via OpenTree API)

```

And here's the mirror repo, which is also missing the four most recent commits:

``` bash
(venv)opentree@ip-10-21-133-161:~/repo/phylesystem-1_par/mirror/phylesystem-1$ git log
commit 823d2aff31d3095dce04696ccc3a8e0d4d58c0b8
Author: Mark T. Holder <mtholder@gmail.com>
Date:   Sat Jan 10 20:47:13 2015 +0000

    rooting and branch length info

    this is tree1 BEAST exponential node dating with calibration densities.

    (Update Study #ot_237 via OpenTree API)

...etc...

```

**Comment by jimallman** [comment]
I saved a tiny change, tailing apache's logs. This all looks reasonable to me, but nothing pushes to GitHub or tickles the webhooks. Maybe it's a celery problem?
```
==> error.log <==
[Sat Jan 24 00:00:52 2015] [error] call_http_json imported
[Sat Jan 24 00:00:52 2015] [error] DEBUG:ot_api.default.global:call_http_json imported

==> access.log <==
67.232.2.77 - - [24/Jan/2015:00:00:52 +0000] "OPTIONS /phylesystem/v1/study/ot_54?author_name=Jim+Allman&author_email=jim%40ibang.com&auth_token=4daeaebe1b36ce9f7747868dc6c4b25bbaaf77ab&starting_commit_SHA=8e54ea98ddd511380c8f6819fc3813d8ab9e9edb&commit_msg=removing+a+tag+(watching+logs) HTTP/1.1" 200 836 "https://tree.opentreeoflife.org/curator/study/edit/ot_54" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.99 Safari/537.36"

==> error.log <==
[Sat Jan 24 00:00:52 2015] [error] call_http_json imported
[Sat Jan 24 00:00:52 2015] [error] DEBUG:ot_api.default.global:call_http_json imported
[Sat Jan 24 00:00:52 2015] [error] DEBUG:ot_api.default.v1.PUT:PUT to study ot_54 for starting_commit_SHA = 8e54ea98ddd511380c8f6819fc3813d8ab9e9edb and merged_SHA = None
[Sat Jan 24 00:00:52 2015] [error] DEBUG:peyotl.nexson_validation._validation_base:detected version as 1.2.1
[Sat Jan 24 00:00:52 2015] [error] DEBUG:peyotl.nexson_validation._validation_base:delete_annotation_from_annot_list with agent_id = peyotl-validator
[Sat Jan 24 00:00:52 2015] [error] DEBUG:peyotl.phylesystem.git_actions:Acquiring lock
[Sat Jan 24 00:00:52 2015] [error] DEBUG:peyotl.phylesystem.git_actions:Found branch "None" for sha "8e54ea98ddd511380c8f6819fc3813d8ab9e9edb"
[Sat Jan 24 00:00:52 2015] [error] DEBUG:peyotl.phylesystem.git_actions:lowest non existing branch =jimallman_study_ot_54_0
[Sat Jan 24 00:00:53 2015] [error] DEBUG:peyotl.phylesystem.git_actions:Created branch "jimallman_study_ot_54_0" with parent "8e54ea98ddd511380c8f6819fc3813d8ab9e9edb"
[Sat Jan 24 00:00:53 2015] [error] DEBUG:peyotl.phylesystem.git_actions:Checked out branch "jimallman_study_ot_54_0"
[Sat Jan 24 00:00:53 2015] [error] DEBUG:peyotl.phylesystem.git_actions:Committed study "ot_54" to branch "jimallman_study_ot_54_0" commit SHA: "bdf0909ac48b6fe9ab4f1c39f0d493854349d4bc"
[Sat Jan 24 00:00:53 2015] [error] DEBUG:peyotl.phylesystem.git_workflows:write of study ot_54 on parent 8e54ea98ddd511380c8f6819fc3813d8ab9e9edb returned = {'prev_file_sha': u'fee4bad3463e201d7ab6370bf667c0ebdc0222f0', 'branch': 'jimallman_study_ot_54_0', 'commit_sha': u'bdf0909ac48b6fe9ab4f1c39f0d493854349d4bc'}
[Sat Jan 24 00:00:53 2015] [error] DEBUG:peyotl.phylesystem.git_workflows:master SHA for that file path is fee4bad3463e201d7ab6370bf667c0ebdc0222f0
[Sat Jan 24 00:00:53 2015] [error] DEBUG:peyotl.phylesystem.git_workflows:merge to master succeeded
[Sat Jan 24 00:00:53 2015] [error] DEBUG:peyotl.phylesystem.git_actions:Releasing lock
[Sat Jan 24 00:00:53 2015] [error] DEBUG:peyotl.phylesystem.git_workflows:returning {'description': 'Updated study #ot_54', 'branch_name': 'master', 'resource_id': 'ot_54', 'sha': u'bdf0909ac48b6fe9ab4f1c39f0d493854349d4bc', 'merge_needed': False, 'error': 0}
[Sat Jan 24 00:00:53 2015] [error] DEBUG:ot_api.default.v1:in __deferred_push_to_gh_call
[Sat Jan 24 00:00:53 2015] [error] DEBUG:ot_api.default.v1:__deferred_push_to_gh_call(https://api.opentreeoflife.org/phylesystem/push/v1/ot_54, {'auth_token': '4daeaebe1b36ce9f7747868dc6c4b25bbaaf77ab'})

==> access.log <==
67.232.2.77 - - [24/Jan/2015:00:00:52 +0000] "PUT /phylesystem/v1/study/ot_54?author_name=Jim+Allman&author_email=jim%40ibang.com&auth_token=4daeaebe1b36ce9f7747868dc6c4b25bbaaf77ab&starting_commit_SHA=8e54ea98ddd511380c8f6819fc3813d8ab9e9edb&commit_msg=removing+a+tag+(watching+logs) HTTP/1.1" 200 2492 "https://tree.opentreeoflife.org/curator/study/edit/ot_54" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.99 Safari/537.36"
```

**Comment by jimallman** [comment]
Progress! Based on @mtholder 's notes in https://github.com/OpenTreeOfLife/phylesystem-api/issues/106, I tailed `/tmp/celeryerr` and found an error related to our SSL certs:
```bash
====
url =https://api.opentreeoflife.org/phylesystem/push/v1/ot_54
verb =PUT
data ={u'auth_token': u'4daeaebe1b36ce9f7747868dc6c4b25bbaaf77ab'}
E1: Traceback (most recent call last):
  File "/home/opentree/repo/phylesystem-api/ot-celery/open_tree_tasks.py", line 31, in call_http_json
    allow_redirects=True)
  File "/home/opentree/venv/local/lib/python2.7/site-packages/requests/api.py", line 49, in request
    response = session.request(method=method, url=url, **kwargs)
  File "/home/opentree/venv/local/lib/python2.7/site-packages/requests/sessions.py", line 461, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/opentree/venv/local/lib/python2.7/site-packages/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "/home/opentree/venv/local/lib/python2.7/site-packages/requests/adapters.py", line 431, in send
    raise SSLError(e, request=request)
SSLError: [Errno 1] _ssl.c:504: error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed

E2: Traceback (most recent call last):
  File "/home/opentree/repo/phylesystem-api/ot-celery/open_tree_tasks.py", line 43, in call_http_json
    x = resp.status_code
AttributeError: 'NoneType' object has no attribute 'status_code'
```
 ... etc ... (but I think the problem is above). We're already chasing this in other issues, but I'll focus on this next since it's obviously blocking data submission and updates to oti.

**Comment by jimallman** [comment]
Here's the main issue regarding our SSL cert validation woes: https://github.com/OpenTreeOfLife/opentree/issues/553

**Comment by jimallman** [comment]
See OpenTreeOfLife/opentree#553 for progress on this issue. The latest test (adding a single .pem with all intermediate certs included) allows us to save study properly. Here's the new output from /etc/celeryerr:
```
url =https://api.opentreeoflife.org/phylesystem/push/v1/ot_54
verb =PUT
data ={u'auth_token': u'4daeaebe1b36ce9f7747868dc6c4b25bbaaf77ab'}

```
... and changes show up promptly in **oti** and in the main study list. I'll mark this as closed once we've finalized these changes in our deployment tools.

**Comment by mtholder** [comment]
thanks for digging through all that, @jimallman I'm sure that was extremely tedious and quite difficult.
I mentioned on that thread that I need to make the celery logs easier to find, and making errors in the push noisier.
I think these were failing in a point that the "failure to push" file was not created. I never saw the web services indicating that pushes were failing, at any rate.  
the whole celery bit of the architecture has too many moving parts. Obviously, it can be made to work, but it would be nice if we could just do that deferred push with threads.   Then any failure would be easy to log in a very obvious way... I'll add that point to  https://github.com/OpenTreeOfLife/phylesystem-api/issues/133

**Comment by jar398** [comment]
Can we close this issue?
